tests/hello_world_comps.c

```diff
@@ -0,0 +1,34 @@
+#include "manager.h"
+#include "limits.h"
+
+extern struct Compartment* loaded_comp;
+
+char*
+get_full_path(char* path)
+{
+    char buf[PATH_MAX];
+    char* res = realpath(path, buf);
+    assert(res != NULL);
+    return res;
+}
+
+int
+main(int argc, char** argv)
+{
+    manager_ddc = cheri_ddc_get();
+    time_t t_buf;
+    time(&t_buf);
+    assert(argc == 2);
+    char* file = argv[1];
+    struct Compartment* hw_comp = comp_from_elf(file);
+    loaded_comp = hw_comp; // TODO
+    hw_comp->argc = 1;
+    char* comp_argv[] = { get_full_path(file) };
```

Comment:
This is related to when I was trying to prepare the stack, which I've since put on indefinite hold. From what I found online [1], the stack is meant to have the arguments upon invoking an executable. However, the Morello ABI [2] seems to deviate from this. I would like to keep this code for now, in case at some point we want to do this "proper".

[1] http://articles.manugarg.com/aboutelfauxiliaryvectors.html
[2] https://git.morello-project.org/morello/kernel/linux/-/wikis/Morello-pure-capability-kernel-user-Linux-ABI-specification

Reply:
I think I understand that (though please correct me if I've completely missed the point). What I meant was why not simply do this?

```c
char *comp_argv[] = { file };  // No get_full_path(...).
```

Reply:
I'm actually unsure. I guess it would work? I've done this in https://github.com/capablevms/CHERI-ELF-comp/pull/1/commits/fa53690fe30750ca708b0cb97dd53295d6628e72 for now.